SDK/backendlessAPI/Classes/Messaging/BackendlessPushHelper.m

```diff
@@ -26,22 +26,28 @@ @implementation BackendlessPushHelper
-    NSURL *fileUrl = [NSURL URLWithString:urlString];
-    [[[NSURLSession sharedSession] downloadTaskWithURL:fileUrl
-                                     completionHandler:^(NSURL *location, NSURLResponse *response, NSError *error) {
-                                         if (location) {
-                                             NSString *tmpDirectory = NSTemporaryDirectory();
-                                             NSString *tmpFile = [[@"file://" stringByAppendingString:tmpDirectory] stringByAppendingString:fileUrl.lastPathComponent];
-                                             NSURL *tmpUrl = [NSURL URLWithString:tmpFile];
-                                             BOOL success = [[NSFileManager defaultManager] moveItemAtURL:location toURL:tmpUrl error:nil];
-                                             UNNotificationAttachment *attachment = [UNNotificationAttachment attachmentWithIdentifier:@"" URL:tmpUrl options:nil error:nil];
-                                             if (attachment) {
-                                                 bestAttemptContent.attachments = @[attachment];
+    
+    if ([request.content.userInfo valueForKey:@"attachment-url"]) {
+       NSString *urlString = [request.content.userInfo valueForKey:@"attachment-url"];
+       NSURL *fileUrl = [NSURL URLWithString:urlString];        
+        [[[NSURLSession sharedSession] downloadTaskWithURL:fileUrl
+                                         completionHandler:^(NSURL *location, NSURLResponse *response, NSError *error) {
+                                             if (location) {
+                                                 NSString *tmpDirectory = NSTemporaryDirectory();
+                                                 NSString *tmpFile = [[@"file://" stringByAppendingString:tmpDirectory] stringByAppendingString:fileUrl.lastPathComponent];
+                                                 NSURL *tmpUrl = [NSURL URLWithString:tmpFile];
+                                                 BOOL success = [[NSFileManager defaultManager] moveItemAtURL:location toURL:tmpUrl error:nil];
+                                                 UNNotificationAttachment *attachment = [UNNotificationAttachment attachmentWithIdentifier:@"" URL:tmpUrl options:nil error:nil];
+                                                 if (attachment) {
+                                                     bestAttemptContent.attachments = @[attachment];
+                                                 }
                                              }
-                                         }
-                                         contentHandler(bestAttemptContent);
-                                     }] resume];
+                                             contentHandler(bestAttemptContent);
+                                         }] resume];
+    }
+    else {
+        contentHandler(bestAttemptContent);
```

Comment:
How about removing the else statement and taking `contentHandler` call out of `if`?

Reply:
The contentHandler(bestAttemptContent); must be called in the downloadTaskWithURL method callback of the current url session (the URLSession class provides an API for downloading content), otherwise the attachment data will be lost.
We have the attachment url and download the attachment from this url to the tmp directory on the device. Then this content should be attached to the bestAttemptContent to present it it in the received push notification.